### PR TITLE
Timeline and bookmark to toucan2

### PR DESCRIPTION
**Pull request by qnkhuat**
Epic https://github.com/metabase/metabase/issues/29231

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/30688)

### CODE REVIEW

File: src/metabase/models/bookmark.clj [hunk 1]
```diff
@@ -8,15 +8,25 @@
...
    [schema.core :as s]
    [toucan.db :as db]
-   [toucan.models :as models]
    [toucan2.core :as t2]))
 
-(models/defmodel CardBookmark :card_bookmark)
-(models/defmodel DashboardBookmark :dashboard_bookmark)
-(models/defmodel CollectionBookmark :collection_bookmark)
-(models/defmodel BookmarkOrdering :bookmark_ordering)
+;; Used to be the toucan1 model name defined using [[toucan.models/defmodel]], not it's a reference to the toucan2 model name.
```
Comment (qnkhuat) [inline src/metabase/models/bookmark.clj:16]:
```suggestion
;; Used to be the toucan1 model name defined using [[toucan.models/defmodel]], now it's a reference to the toucan2 model name.
```

File: src/metabase/models/timeline.clj [hunk 2]
```diff
@@ -2,19 +2,27 @@
...
    [metabase.util.date-2 :as u.date]
+   [methodical.core :as methodical]
    [schema.core :as s]
    [toucan.hydrate :refer [hydrate]]
-   [toucan.models :as models]
    [toucan2.core :as t2]))
 
-(models/defmodel Timeline :timeline)
+(def Timeline
+  "Used to be the toucan1 model name defined using [[toucan.models/defmodel]], not it's a reference to the toucan2 model name.
```
Comment (qnkhuat) [inline src/metabase/models/timeline.clj:15]:
```suggestion
  "Used to be the toucan1 model name defined using [[toucan.models/defmodel]], now it's a reference to the toucan2 model name.
```

File: src/metabase/models/timeline_event.clj [hunk 3]
```diff
@@ -4,14 +4,21 @@
...
    [metabase.util.honey-sql-2 :as h2x]
+   [methodical.core :as methodical]
    [schema.core :as s]
    [toucan.hydrate :refer [hydrate]]
-   [toucan.models :as models]
    [toucan2.core :as t2]))
 
-(models/defmodel TimelineEvent :timeline_event)
+(def TimelineEvent
+  "Used to be the toucan1 model name defined using [[toucan.models/defmodel]], not it's a reference to the toucan2 model name.
```
Comment (qnkhuat) [inline src/metabase/models/timeline_event.clj:13]:
```suggestion
  "Used to be the toucan1 model name defined using [[toucan.models/defmodel]], now it's a reference to the toucan2 model name.
```

File: src/metabase/models/bookmark.clj [hunk 4]
```diff
@@ -8,15 +8,25 @@
...
+(def CollectionBookmark "CollectionBookmark model" :model/CollectionBookmark)
+(def BookmarkOrdering   "BookmarkOrdering model"   :model/BookmarkOrdering)
+
+(methodical/defmethod t2/table-name :model/CardBookmark       [_model] :card_bookmark)
+(methodical/defmethod t2/table-name :model/DashboardBookmark  [_model] :dashboard_bookmark)
+(methodical/defmethod t2/table-name :model/CollectionBookmark [_model] :collection_bookmark)
+(methodical/defmethod t2/table-name :model/BookmarkOrdering   [_model] :bookmark_ordering)
+
+(doseq [m [:model/CardBookmark :model/DashboardBookmark :model/CollectionBookmark :model/BookmarkOrdering]]
+  (derive m :metabase/model))
```
Comment (bshepherdson) [inline src/metabase/models/bookmark.clj:29]: nit: maybe this is just my tastes, but for a one-liner like this I'd rather see it unrolled as four `derive` calls.

**Review by bshepherdson — APPROVED** [review]

**Pull request merged**